src/library/scala/collection/immutable/Seq.scala

```diff
@@ -34,6 +34,28 @@ trait IndexedSeq[+A] extends Seq[A]
 
   final override def toIndexedSeq: IndexedSeq[A] = this
 
+  override def canEqual(that: Any): Boolean = that match {
+    case otherIndexedSeq: IndexedSeq[_] => length == otherIndexedSeq.length && super.canEqual(that)
+    case _ => super.canEqual (that)
+  }
+
+  override def equals(o: Any): Boolean = o match {
+    case otherIndexedSeq: IndexedSeq[_] =>
+      (this eq otherIndexedSeq) || {
+        var equal = length == otherIndexedSeq.length
```

Comment:
I'd rather use a `return` in the loop instead of messing around with a mutable variable but I suppose it's a matter of style and both should work fine.